### PR TITLE
Migrate pkg/kubelet/config to structured logging

### PR DESCRIPTION
**Pull request by LittleGoWitch**
#### What type of PR is this?

/kind cleanup
/sig instrumentation
/sig node
/priority important-soon

#### What this PR does / why we need it:
Migrate `pkg/kubelet/config` to structured logging
#### Which issue(s) this PR fixes:

Ref #98976
#### Special notes for your reviewer:
part of [kubernetes/enhancements#1602](https://github.com/kubernetes/enhancements/issues/1602)
As it is related to log, change the log here to structured log.
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

- [KEP]: [Structured Logging](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging)

```docs

```

### CODE REVIEW

**Comment by shubheksha** [comment]
Hey @AfrouzMashayekhi, we normally don't use full stops after error messages. Logging both the object and the specific info about the same object seems repetitive. I'd also advice not modifying the error messages too much unless to clarify them further. If you could address this across your PR, I'm happy to take another look.

**Comment by AfrouzMashayekhi** [comment]
@shubheksha hope it's ok now

File: pkg/kubelet/config/file_linux.go [hunk 1]
```diff
@@ -102,7 +102,7 @@ func (s *sourceFile) doWatch() error {
 func (s *sourceFile) produceWatchEvent(e *fsnotify.Event) error {
 	// Ignore file start with dots
 	if strings.HasPrefix(filepath.Base(e.Name), ".") {
-		klog.V(4).Infof("Ignored pod manifest: %s, because it starts with dots", e.Name)
+		klog.V(4).InfoS("Ignored pod manifest, because it starts with dots", "event", e.Name)
```
Comment (shubheksha) [inline pkg/kubelet/config/file_linux.go:105]: This is missing an "eventName" key

File: pkg/kubelet/config/http.go [hunk 2]
```diff
@@ -63,16 +63,16 @@ func (s *sourceURL) run() {
...
+			klog.InfoS("Failed to read pods from URL. Dropping verbosity of this message to V(4)", "err", err)
 		} else {
-			klog.V(4).Infof("Failed to read pods from URL: %v", err)
+			klog.V(4).InfoS("Failed to read pods from URL", "err", err)
 		}
 		s.failureLogs++
 	} else {
 		if s.failureLogs > 0 {
-			klog.Info("Successfully read pods from URL.")
+			klog.InfoS("Successfully read pods from URL.")
```
Comment (shubheksha) [inline pkg/kubelet/config/http.go:75]:
```suggestion
			klog.InfoS("Successfully read pods from URL")
```

File: pkg/kubelet/config/file.go [hunk 3]
```diff
@@ -173,32 +173,32 @@ func (s *sourceFile) extractFromDir(name string) ([]*v1.Pod, error) {
...
+			klog.ErrorS(nil, "Manifest path is not a directory or file", "path", path, "mode", statInfo.Mode())
 		}
 	}
 	return pods, nil
 }
 
 // extractFromFile parses a file for Pod configuration information.
 func (s *sourceFile) extractFromFile(filename string) (pod *v1.Pod, err error) {
-	klog.V(3).Infof("Reading config file %q", filename)
+	klog.V(3).InfoS("Read config file", "file", filename)
```
Comment (shubheksha) [inline pkg/kubelet/config/file.go:201]:
```suggestion
	klog.V(3).InfoS("Read config file", "fileName", filename)
```

File: pkg/kubelet/config/file.go [hunk 4]
```diff
@@ -173,32 +173,32 @@ func (s *sourceFile) extractFromDir(name string) ([]*v1.Pod, error) {
...
 				if !os.IsNotExist(err) {
-					klog.Errorf("Can't process manifest file %q: %v", path, err)
+					klog.ErrorS(err, "Could not process manifest file", "path", path)
 				}
 			} else {
 				pods = append(pods, pod)
 			}
 		default:
-			klog.Errorf("Manifest path %q is not a directory or file: %v", path, statInfo.Mode())
+			klog.ErrorS(nil, "Manifest path is not a directory or file", "path", path, "mode", statInfo.Mode())
```
Comment (shubheksha) [inline pkg/kubelet/config/file.go:193]: Not sure we should be using `ErrorS` here as there isn't an error. Perhaps `InfoS`? @serathius what do you think?

File: pkg/kubelet/config/file.go [hunk 5]
```diff
@@ -173,32 +173,32 @@ func (s *sourceFile) extractFromDir(name string) ([]*v1.Pod, error) {
...
 		if err != nil {
-			klog.Errorf("Can't get metadata for %q: %v", path, err)
+			klog.ErrorS(err, "Could not get metadata", "path", path)
 			continue
 		}
 
 		switch {
 		case statInfo.Mode().IsDir():
-			klog.Errorf("Not recursing into manifest path %q", path)
+			klog.ErrorS(nil, "Not recursing into manifest path", "path", path)
```
Comment (shubheksha) [inline pkg/kubelet/config/file.go:182]: Not sure we should be using ErrorS here as there isn't an error.

File: pkg/kubelet/config/file.go [hunk 6]
```diff
@@ -65,7 +65,7 @@ func NewSourceFile(path string, nodeName types.NodeName, period time.Duration, u
 	path = strings.TrimRight(path, string(os.PathSeparator))
 
 	config := newSourceFile(path, nodeName, period, updates)
-	klog.V(1).Infof("Watching path %q", path)
+	klog.V(1).InfoS("Watching path.", "path", path)
```
Comment (shubheksha) [inline pkg/kubelet/config/file.go:68]:
```suggestion
	klog.V(1).InfoS("Watching path", "path", path)
```

File: pkg/kubelet/config/config.go [hunk 7]
```diff
@@ -94,7 +94,7 @@ func (c *PodConfig) SeenAllSources(seenSources sets.String) bool {
 	if c.pods == nil {
 		return false
 	}
-	klog.V(5).Infof("Looking for %v, have seen %v", c.sources.List(), seenSources)
+	klog.V(5).InfoS("Looking for source list, have seen source ", "sourceList", c.sources.List(), "seenSources", seenSources)
```
Comment (shubheksha) [inline pkg/kubelet/config/config.go:97]:
```suggestion
	klog.V(5).InfoS("Looking for list of sources, have seen source ", "sourceList", c.sources.List(), "seenSources", seenSources)
```

File: pkg/kubelet/config/common.go [hunk 8]
```diff
@@ -68,16 +68,16 @@ func applyDefaults(pod *api.Pod, source string, isFile bool, nodeName types.Node
...
 
 	pod.Name = generatePodName(pod.Name, nodeName)
-	klog.V(5).Infof("Generated Name %q for UID %q from URL %s", pod.Name, pod.UID, source)
+	klog.V(5).InfoS("Generated Name for UID from URL", "pod", klog.KObj(pod), "podUID", pod.UID, "source", source)
 
 	if pod.Namespace == "" {
 		pod.Namespace = metav1.NamespaceDefault
 	}
-	klog.V(5).Infof("Using namespace %q for pod %q from %s", pod.Namespace, pod.Name, source)
+	klog.V(5).InfoS("Used namespace for pod from source", "pod", klog.KObj(pod), "source", source)
```
Comment (shubheksha) [inline pkg/kubelet/config/common.go:81]: Why are we logging the full object here?

**Review by shubheksha — CHANGES_REQUESTED** [review]

File: pkg/kubelet/config/common.go (hunk 8 above)
Comment (AfrouzMashayekhi) [reply pkg/kubelet/config/common.go:81]: In the [KEP](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#use-klogkobj-and-klogkref-for-kubernetes-objects) guide said we should reference Kubernetes objects.

File: pkg/kubelet/config/file.go (hunk 5 above)
Comment (AfrouzMashayekhi) [reply pkg/kubelet/config/file.go:182]: we shouldn't change log level based on [this comment](https://github.com/kubernetes/kubernetes/issues/98976#issuecomment-791734797)

File: pkg/kubelet/config/config.go [hunk 9]
```diff
@@ -94,7 +94,7 @@ func (c *PodConfig) SeenAllSources(seenSources sets.String) bool {
 	if c.pods == nil {
 		return false
 	}
-	klog.V(5).Infof("Looking for %v, have seen %v", c.sources.List(), seenSources)
+	klog.V(5).InfoS("Looking for list of sources, have seen source ", "sourceList", c.sources.List(), "seenSources", seenSources)
```
Comment (serathius) [inline pkg/kubelet/config/config.go:97]:
```suggestion
	klog.V(5).InfoS("Looking for list of sources, have seen source ", "sources", c.sources.List(), "seenSources", seenSources)
```

File: pkg/kubelet/config/file.go [hunk 10]
```diff
@@ -173,32 +173,32 @@ func (s *sourceFile) extractFromDir(name string) ([]*v1.Pod, error) {
...
+			klog.ErrorS(nil, "Manifest path is not a directory or file", "path", path, "mode", statInfo.Mode())
 		}
 	}
 	return pods, nil
 }
 
 // extractFromFile parses a file for Pod configuration information.
 func (s *sourceFile) extractFromFile(filename string) (pod *v1.Pod, err error) {
-	klog.V(3).Infof("Reading config file %q", filename)
+	klog.V(3).InfoS("Read config file", "fileName", filename)
```
Comment (serathius) [inline pkg/kubelet/config/file.go:201]:
```suggestion
	klog.V(3).InfoS("Read config file", "path", filename)
```

File: pkg/kubelet/config/file.go (hunk 4 above)
Comment (serathius) [reply pkg/kubelet/config/file.go:193]: From migration guide
`Calling ErrorS with nil as error is semi-acceptable if there is error condition that deserves a stack trace at this origin point. For expected errors (errors that can happen during routine operations) please consider using klog.InfoS and pass error in err key instead.`
I don't think this is expected situation and using `ErrorS` with nil is ok here.

File: pkg/kubelet/config/common.go (hunk 8 above)
Comment (serathius) [reply pkg/kubelet/config/common.go:81]: `klog.KObj` doesn't log whole object, just a reference in standard format `<namespace>/<name>`. Main reason for using it is it both supports `text` and `json` format.

File: pkg/kubelet/config/config.go [hunk 11]
```diff
@@ -254,16 +254,16 @@ func (s *podStorage) merge(source string, change interface{}) (adds, updates, de
 	switch update.Op {
 	case kubetypes.ADD, kubetypes.UPDATE, kubetypes.DELETE:
 		if update.Op == kubetypes.ADD {
-			klog.V(4).Infof("Adding new pods from source %s : %v", source, update.Pods)
+			klog.V(4).InfoS("Added new pods from source", "source", source, "pods", update.Pods)
```
Comment (serathius) [inline pkg/kubelet/config/config.go:257]: Could you use `format.Pods` from https://github.com/kubernetes/kubernetes/pull/99799

File: pkg/kubelet/config/file.go [hunk 12]
```diff
@@ -173,32 +173,32 @@ func (s *sourceFile) extractFromDir(name string) ([]*v1.Pod, error) {
...
+			klog.ErrorS(nil, "Manifest path is not a directory or file", "path", path, "mode", statInfo.Mode())
 		}
 	}
 	return pods, nil
 }
 
 // extractFromFile parses a file for Pod configuration information.
 func (s *sourceFile) extractFromFile(filename string) (pod *v1.Pod, err error) {
-	klog.V(3).Infof("Reading config file %q", filename)
+	klog.V(3).InfoS("Read config file", "path", filename)
```
Comment (ehashman) [inline pkg/kubelet/config/file.go:201]:
```suggestion
	klog.V(3).InfoS("Reading config file", "path", filename)
```

**Review by ehashman — COMMENTED** [review]
/ok-to-test
/triage accepted

File: pkg/kubelet/config/file.go (hunk 3 above)
Comment (kikisdeliveryservice) [reply pkg/kubelet/config/file.go]: this change wasn't made

File: pkg/kubelet/config/config.go (hunk 7 above)
Comment (kikisdeliveryservice) [reply pkg/kubelet/config/config.go]: suggested change not implemented

File: pkg/kubelet/config/config.go (hunk 9 above)
Comment (kikisdeliveryservice) [reply pkg/kubelet/config/config.go]: suggested change conflict with one above (sources vs sourceList), but both agree on text ("Looking for list of sources") which is currently different ("Looking for sources")

File: pkg/kubelet/config/file.go (hunk 3 above)
Comment (kikisdeliveryservice) [reply pkg/kubelet/config/file.go]: looking closer at this func() extractFromFile takes a path: https://github.com/kubernetes/kubernetes/blob/cea1d4e20b4a7886d8ff65f34c6d4f95efcb4742/pkg/kubelet/config/file.go#L184

so current iteration is correct.

**Review by kikisdeliveryservice — COMMENTED** [review]
Verified requested changes made, review changes, checked files for completeness.  Passing this to approvers

/lgtm

**Review by mrunalp — APPROVED** [review]

**Pull request merged**